### PR TITLE
Consume new ID Crypt SDK

### PR DESCRIPTION
**Pull request by DavidRamsbottomRTGS**

### CODE REVIEW

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/HttpHandlers/StatusCodeHttpHandler.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage reques
 
 		var response = new HttpResponseMessage(responseMock.HttpStatusCode)
 		{
-			Content = responseMock.Content
+			Content = responseMock.Content != null ? new StringContent(responseMock.Content) : null
```
Comment (Bassetts) [inline src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/HttpHandlers/StatusCodeHttpHandler.cs:31]: Same suggestion here, only assign when `responseMock.Content` is not null

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Publisher/TestData/MockHttpResponse.cs [hunk 2]
```diff
@@ -1,11 +1,10 @@
 ﻿using System.Net;
-using System.Net.Http;
 
 namespace RTGS.DotNetSDK.IntegrationTests.Publisher.TestData;
 
 internal record MockHttpResponse
 {
 	public string Path { get; init; }
```
Comment (Bassetts) [inline src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Publisher/TestData/MockHttpResponse.cs:7]: This isn't used anywhere after out refactoring of the status code handlers so could be removed like we did in the IDC SDK

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK/Subscriber/Handlers/Internal/IdCryptBankInvitationV1Handler.cs [hunk 3]
```diff
@@ -127,36 +136,33 @@ private async Task<ConnectionAccepted> PollConnectionState(string connectionId)
...
 	}
 
-	private async Task<ConnectionAccepted> GetConnection(string connectionId)
+	private async Task<ConnectionResponse> GetConnection(string connectionId)
 	{
 		try
 		{
-			_logger.LogDebug("Sending GetConnection request to ID Crypt with connection Id '{ConnectionId}'",
+			_logger.LogDebug("Sending GetConnection request to ID Crypt",
 				connectionId);
```
Comment (Bassetts) [inline src/RTGS.DotNetSDK/RTGS.DotNetSDK/Subscriber/Handlers/Internal/IdCryptBankInvitationV1Handler.cs:144]: `connectionId` is not used

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK/Subscriber/Handlers/Internal/IdCryptBankInvitationV1Handler.cs [hunk 4]
```diff
@@ -127,36 +136,33 @@ private async Task<ConnectionAccepted> PollConnectionState(string connectionId)
...
 
-			if (connection is null)
-			{
-				throw new RtgsSubscriberException(
-					$"Unexpected null value returned when sending GetConnection request to ID Crypt with connection Id '{connectionId}'");
-			}
-
-			_logger.LogDebug("Sent GetConnection request to ID Crypt with connection Id '{ConnectionId}'",
+			_logger.LogDebug("Sent GetConnection request to ID Crypt",
 				connectionId);
```
Comment (Bassetts) [inline src/RTGS.DotNetSDK/RTGS.DotNetSDK/Subscriber/Handlers/Internal/IdCryptBankInvitationV1Handler.cs:149]: `connectionId` is not used

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK/Subscriber/Handlers/Internal/IdCryptCreateInvitationRequestV1Handler.cs [hunk 5]
```diff
@@ -64,22 +68,31 @@ private async Task<ConnectionInviteResponseModel> CreateIdCryptInvitationAsync(s
...
 			_logger.LogDebug("Sending CreateInvitation request with alias {Alias} to ID Crypt Cloud Agent", alias);
 
-			var response = await _identityClient.Connection.CreateInvitation(
+			var response = await _connectionsClient.CreateInvitationAsync(
 				alias,
 				autoAccept,
 				multiUse,
-				usePublicDid);
+				usePublicDid,
+				default);
```
Comment (Bassetts) [inline src/RTGS.DotNetSDK/RTGS.DotNetSDK/Subscriber/Handlers/Internal/IdCryptCreateInvitationRequestV1Handler.cs:76]: `default` is not needed here.

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Subscriber/InternalHandlers/GivenIdCryptBankInvitationSentToOpenSubscriberConnection/AndIdCryptApiAvailable.cs [hunk 6]
```diff
@@ -251,20 +251,20 @@ public async Task WhenCallingIdCryptAgent_ThenLog()
 			_bankInvitationNotificationHandler.WaitForMessage(WaitForReceivedMessageDuration);
 
 			var bankDid = ValidMessages.IdCryptBankInvitationV1.FromBankDid;
-			var connectionId = ReceiveInvitation.Response.ConnectionID;
+			var connectionId = ReceiveInvitation.Response.ConnectionId;
```
Comment (Bassetts) [inline src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Subscriber/InternalHandlers/GivenIdCryptBankInvitationSentToOpenSubscriberConnection/AndIdCryptApiAvailable.cs:254]: `connectionId` is not used

**Review by Bassetts — CHANGES_REQUESTED** [review]

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/HttpHandlers/QueueableStatusCodeHttpHandler.cs [hunk 7]
```diff
@@ -28,7 +28,7 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage reques
 
 		var response = new HttpResponseMessage(responseMock.HttpStatusCode)
 		{
-			Content = responseMock.Content
+			Content = responseMock.Content != null ? new StringContent(responseMock.Content) : null
```
[the remark this answers is not in the document]
Comment (DavidRamsbottomRTGS) [reply src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/HttpHandlers/QueueableStatusCodeHttpHandler.cs]: Done (kinda)

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/HttpHandlers/StatusCodeHttpHandler.cs (hunk 1 above)
Comment (DavidRamsbottomRTGS) [reply src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/HttpHandlers/StatusCodeHttpHandler.cs:31]: Done (kinda)

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK/Subscriber/Handlers/Internal/IdCryptBankInvitationV1Handler.cs (hunk 4 above)
Comment (DavidRamsbottomRTGS) [reply src/RTGS.DotNetSDK/RTGS.DotNetSDK/Subscriber/Handlers/Internal/IdCryptBankInvitationV1Handler.cs:149]: Done

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK/Subscriber/Handlers/Internal/IdCryptCreateInvitationRequestV1Handler.cs (hunk 5 above)
Comment (DavidRamsbottomRTGS) [reply src/RTGS.DotNetSDK/RTGS.DotNetSDK/Subscriber/Handlers/Internal/IdCryptCreateInvitationRequestV1Handler.cs:76]: Done

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Subscriber/InternalHandlers/GivenIdCryptBankInvitationSentToOpenSubscriberConnection/AndIdCryptApiAvailable.cs (hunk 6 above)
Comment (DavidRamsbottomRTGS) [reply src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Subscriber/InternalHandlers/GivenIdCryptBankInvitationSentToOpenSubscriberConnection/AndIdCryptApiAvailable.cs:254]: Done

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Subscriber/InternalHandlers/GivenIdCryptBankInvitationSentToOpenSubscriberConnection/AndIdCryptApiAvailable.cs [hunk 8]
```diff
@@ -618,22 +618,22 @@ public async Task WhenCallingIdCryptAgent_ThenLog()
...
```
[the remark this answers is not in the document]
Comment (DavidRamsbottomRTGS) [reply src/RTGS.DotNetSDK/RTGS.DotNetSDK.IntegrationTests/Subscriber/InternalHandlers/GivenIdCryptBankInvitationSentToOpenSubscriberConnection/AndIdCryptApiAvailable.cs:621]: Done

**Review by Bassetts — APPROVED** [review]

**Pull request merged**